### ISSUE TITLE
TypeError: _this2.print is not a function

### ISSUE THREAD

**Issue by GabrielTK** [maierfelix/POGOserver#225]
Good Evening,
I'm trying to run this server, but i'm getting the following error:
```
[Console] someone*****@gmail.com connected!
[Console] Unhandled exception occurred:
[Console] TypeError: _this2.print is not a function
[Console] TypeError: _this2.print is not a function
    at Query._callback (C:/NodeJS/Projects/POGOServer/Core/src/models/World/players.js:128:26)
    at Query.Sequence.end (C:\NodeJS\Projects\POGOServer\Core\node_modules\mysql\lib\protocol\sequences\Sequence.js:85:24)
    at Query.ErrorPacket (C:\NodeJS\Projects\POGOServer\Core\node_modules\mysql\lib\protocol\sequences\Query.js:94:8)

    at Protocol._parsePacket (C:\NodeJS\Projects\POGOServer\Core\node_modules\mysql\lib\protocol\Protocol.js:280:23)
    at Parser.write (C:\NodeJS\Projects\POGOServer\Core\node_modules\mysql\lib\protocol\Parser.js:74:12)
    at Protocol.write (C:\NodeJS\Projects\POGOServer\Core\node_modules\mysql\lib\protocol\Protocol.js:39:16)
    at Socket.<anonymous> (C:\NodeJS\Projects\POGOServer\Core\node_modules\mysql\lib\Connection.js:109:28)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:176:18)
    at Socket.Readable.push (_stream_readable.js:134:10)
    at TCP.onread (net.js:543:20)
[Console] The server has crashed!
```

Can anyone help me please??
Thank you,
Gabriel

**Comment by atomboy2005** [comment]
Try reinstalling you must of edited one of the downloaded files

**Comment by GabrielTK** [comment]
Ok, I will try it now. 
NOTE: I'm using fiddler with a script.

**Comment by atomboy2005** [comment]
@GabrielTK What OS are you on?

**Comment by GabrielTK** [comment]
Windows 10.

**Comment by GabrielTK** [comment]
This is what i Added to `OnBeforeRequest` on the Fiddler Script:
```if (oSession.uriContains("pgorelease.nianticlabs.com/plfe/rpc")) {
            oSession["ui-color"] = "red";
            oSession.host = "192.168.0.20";
            oSession.port=3000;
            oSession.oRequest.headers.UriScheme = "http";```

**Comment by atomboy2005** [comment]
did it work?

**Comment by GabrielTK** [comment]
Dumping iOS model pm0132

**Comment by atomboy2005** [comment]
Looking at the error it seems like it worked

**Comment by GabrielTK** [comment]
Going to test it now!

**Comment by GabrielTK** [comment]
Nope

**Comment by atomboy2005** [comment]
Hmmmm try building it via docker

**Comment by atomboy2005** [comment]
that's what worked for me

**Comment by GabrielTK** [comment]
I think it worked!! (Through Windows)

**Comment by atomboy2005** [comment]
Nice

**Comment by GabrielTK** [comment]
No, it isn't.
But a different error:
```
[Console] Unhandled exception occurred:
[Console] TypeError: Cannot read property 'contents' of undefined
[Console] TypeError: Cannot read property 'contents' of undefined
    at GameServer.authenticatePlayer (C:/NodeJS/Projects/POGOServer/Core/src/request.js:190:31)
    at GameServer.onRequest (C:/NodeJS/Projects/POGOServer/Core/src/request.js:97:10)
    at GameServer.processRpcRequest (C:/NodeJS/Projects/POGOServer/Core/src/request.js:84:10)
    at GameServer.routeRequest (C:/NodeJS/Projects/POGOServer/Core/src/request.js:28:12)
    at IncomingMessage.<anonymous> (C:/NodeJS/Projects/POGOServer/Core/src/http.js:22:12)
    at emitNone (events.js:86:13)
    at IncomingMessage.emit (events.js:185:7)
    at endReadableNT (_stream_readable.js:974:12)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickDomainCallback (internal/process/next_tick.js:122:9)
[Console] The server has crashed!
```

**Comment by GabrielTK** [comment]
So... Let's give docker a try.

**Comment by atomboy2005** [comment]
Docker does take a while

**Comment by atomboy2005** [comment]
10-20 mins

**Comment by GabrielTK** [comment]
Ok. Let's give it a try...

**Comment by atomboy2005** [comment]
good luck

**Comment by GabrielTK** [comment]
But wait: Do the assets are the same as the PC version??

**Comment by atomboy2005** [comment]
yeah

**Comment by GabrielTK** [comment]
So... I can try to copy them to Docker!!! So we won't need to wait 2 years 😆

**Comment by atomboy2005** [comment]
It's worth a try, but since docker uses the linux library I think you will have download all the required apps since docker is independent from windows so docker will need some software to run, have the dockerfile do all that for you

**Comment by GabrielTK** [comment]
Ok, understood. Setting it up now.

**Comment by GabrielTK** [comment]
Copied the data folder from the PC to the Docker Container folder, and added `COPY data /POGOserver/data` to the Dockerfile

**Comment by atomboy2005** [comment]
yeah the app models will work fine it's just the modules that run the server won't work since you haven't runned the dockerfile

**Comment by GabrielTK** [comment]
Ok, so now docker uses Hyper-V. Seriously?? I wont be able to use my VMWare Workstation anymore!! 😭 Sorry Docker, but I won't update you now.

**Comment by GabrielTK** [comment]
Ok, I've run `docker build -t pogoserver .` and it build sucessfully. So, what do I do now???

**Comment by GabrielTK** [comment]
Ok, I ran it in Docker. I constantly get the ’print2’ error. Any idea on what to do??

**Comment by GabrielTK** [comment]
Solved at PR #238
